### ISSUE TITLE
'tson::World': no appropriate default constructor available

### ISSUE THREAD

**Issue by DomRe** [SSBMTonberry/tileson#74]
Occurs in ```ProjectFolder.hpp``` at line 32.

Building on Vs 2022 17.3.6 using C++20 flag.

**Comment by DomRe** [comment]
Code is included directly into project, no preprocessor symbols defined.

**Comment by DomRe** [comment]
I also noticed that there are several includes for each of the different JSON types, resulting in duplicate json libraries. I would suggest removing these from the main library and leaving it up to the user to implement, but keeping the examples of course, but separate.

**Comment by DomRe** [comment]
This error appears to occur due to removing the json11 parser in favor of a different one.

**Comment by SSBMTonberry** [comment]
As I understand it you use `tileson_min.hpp` with an own parser? :stuck_out_tongue: 
Any reason why you're not using the default parser provided in `tileson.hpp`?

I will also need some code and some more context to be able to help you out here. Using `C++20` should be fine. I'm in fact using `Tileson` in a `C++20` project myself, though I'm using `Linux` and `GCC`.

**Comment by DomRe** [comment]
@SSBMTonberry 

Ah I see. I will use ```tileson_min.hpp``` going forward.

I already use ```nlohmann::json``` in my project, which is why I wanted to use that over ```json11```.

**Comment by SSBMTonberry** [comment]
Right. That should work just fine, but you might as well use `tileson_min.hpp` when using your own parser :)
Just did a small test locally for world loading. Maybe that gives you an idea how to solve your problem :slightly_smiling_face: 

The `#include "../include/external/nlohmann.hpp"` would be whatever version you use of `nlohmann json parser`, but `#include "../include/json/NlohmannJson.hpp"` will be required to be able to use the `NlohmannJson` overload of the internal `tson::IJson`. This file is not included in `tileson.hpp` nor `tileson_min.hpp`, but it's located here:
 - https://github.com/SSBMTonberry/tileson/blob/master/include/json/NlohmannJson.hpp

Some example code I used for my internal test I hope will work :slightly_smiling_face: 
```c++
#include "../tileson.hpp"
#include "../external_libs/catch.hpp"
#include "../include/external/nlohmann.hpp"
#include "../include/json/NlohmannJson.hpp"

TEST_CASE( "Nlohmann: Parse world - Expect 4 maps and parsed data", "[project][world]" )
{
    fs::path pathLocal {"../../content/test-maps/project/world/test.world"};
    fs::path pathTravis {"../content/test-maps/project/world/test.world"};
    fs::path pathToUse = (fs::exists(pathLocal)) ? pathLocal : pathTravis;

    tson::World world {pathToUse, std::make_unique<tson::NlohmannJson>()};
    REQUIRE(world.getMapData().size() == 4);
    REQUIRE(world.get("not_exists") == nullptr);
    REQUIRE(world.get("w1.json") != nullptr);
    REQUIRE(world.get("w1.json")->size == tson::Vector2i (256, 128));
    REQUIRE(world.get("w1.json")->position == tson::Vector2i (-256, -128));
    REQUIRE(!world.onlyShowAdjacentMaps());
    REQUIRE(world.getType() == "world");
}
```
